src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -127,8 +127,20 @@ int BoutMesh::load() {
   output << "\tGrid size: " << nx << " x " << ny << " x " << MZ << endl;
 
   // Get guard cell sizes
-  options->get("MXG", MXG, 2);
-  options->get("MYG", MYG, 2);
+  // Try to read from grid file first, then if not found
+  // get from options
+  if (Mesh::get(MXG, "MXG")) {
+    // Error code returned
+    options->get("MXG", MXG, 2);
+  }
+  ASSERT0(MXG >= 0);
```

Comment:
Sorry, missed the declaration of `CHECKLEVEL` above the definition.
So this is only disabled if somebody declares `-DCHECK=-1` 👍